src/macros.rs

```diff
@@ -62,17 +62,54 @@ macro_rules! const_mat2 {
 /// ```
 #[macro_export]
 macro_rules! const_mat3 {
-    ($f32x9:expr) => {
-        unsafe { $crate::f32::F32x9Cast { f32x9: $f32x9 }.mat3 }
-    };
-    ($col0:expr, $col1:expr, $col2:expr) => {
+    ($f32x9:expr) => {{
+        #[cfg(not(target_arch = "spirv"))]
+        unsafe {
+            $crate::f32::F32x9Cast { f32x9: $f32x9 }.mat3
+        }
+
+        // special path for SPIRV, without this we get:
+        //
+        //    error[E0080]: it is undefined behavior to use this value
+        //    --> glam-rs\src\f32\mat3.rs:6:1
+        //     |
+        //   6 | const ZERO: Mat3 = const_mat3!([0.0; 9]);
+        //     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered uninitialized bytes at .z_axis.1, but expected initialized plain (non-pointer) bytes
+        //     |
+        //     = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rustc repository if you believe it should not be considered undefined behavior.
+        #[cfg(target_arch = "spirv")]
+        $crate::Mat3 {
```

Comment:
This should work for all platforms shouldn't it? I wrote these macros before the members were public so it wouldn't have been possible to do it this way. It might be cleaner to do:

```
/// Creates a `Mat3` from three column vectors that can be used to initialize a constant value.
///
/// ```
/// # #[macro_use] extern crate glam;
/// use glam::{const_mat3, Mat3};
/// const ZERO: Mat3 = const_mat3!([0.0; 9]);
/// const IDENTITY: Mat3 = const_mat3!([1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]);
/// ```
#[macro_export]
macro_rules! const_mat3 {
    ($f32x9:expr) => {
        Mat3 {
            x_axis: const_vec3!([$f32x9[0], $f32x9[1], $f32x9[2]]),
            y_axis: const_vec3!([$f32x9[3], $f32x9[4], $f32x9[5]]),
            z_axis: const_vec3!([$f32x9[6], $f32x9[7], $f32x9[8]]),
        }
    };
    ($col0:expr, $col1:expr, $col2:expr) => {
        Mat3 {
            x_axis: const_vec3!($col0),
            y_axis: const_vec3!($col1),
            z_axis: const_vec3!($col2),
        }
    };
}
```

Either way I think it should be possible to use the same technique for spriv and non-spriv.